### DISCUSSION TITLE
libssh2_session handshake fails on VMware ESXi

### DISCUSSION THREAD

**Discussion by casperjSUM** [libssh2/libssh2#1069] [Q&A]
**Describe the bug**
I'm trying to use libssh2 on a VMware ESXi system. libssh2 was built as a static library and linked to an ESXi application. After I built and ran the example/ssh2_exec on ESXi, the call to libssh2_session_handshake failed with the error "Failure establishing SSH session: (-1) Unable to exchange encryption keys"

**To Reproduce**
Simply build and run the ssh2_exec example code on a VMware ESXi system. The code snippet is shown below:

```c
    /* Create a session instance */
    session = libssh2_session_init();
    if(!session)
        return -1;

    /* tell libssh2 we want it all done non-blocking */
    libssh2_session_set_blocking(session, 0);

    /* ... start it up. This will trade welcome banners, exchange keys,
     * and setup crypto, compression, and MAC layers
     */
    while((rc = libssh2_session_handshake(session, sock)) ==
           LIBSSH2_ERROR_EAGAIN);
    if(rc) {
        char* errmsg;
        int errlen;
        int err = libssh2_session_last_error(session, &errmsg, &errlen, 0);
        fprintf(stderr, "Failure establishing SSH session: (%d) %s\n", err, errmsg);
        return -1;
    }
```

Run the example code, the output is:
`Failure establishing SSH session: (-1) Unable to exchange encryption keys`

**Expected behavior**
Run the same code on other Linux systems such as Ubuntu, Redhat, and CentOS. The SSH connection can be established and the remote command can be executed successfully.

**Version (please complete the following information):**
 - local OS and version: VMware ESXi 7.0.0
 - remote host OS: Any 
 - libssh2 version: 1.10.0
 - crypto backend: OpenSSL 1.1.1q

**Additional context**
Because the SSH shell command works and can connect to a remote server, I assume that it's not a system configuration issue.

**Comment by PengShiJin** [comment]
Hi, did you solve the problem? I am having the inability to exchange keys when running libssh2 on an ARM development board. But I can't find a solution.
